src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -2563,9 +2577,9 @@ private Expression<Action<FunctionContext>> CompileNamedBlock(NamedBlockAst name
             // the return from the trap must return from the function containing the trap.  So we wrap the full
             // body of regular begin/process/end blocks with a try/catch so a return from the trap returns
             // to the right place.  We can avoid also avoid generating the catch if we know there aren't any traps.
-            if (!_compilingTrap &&
-                ((traps != null && traps.Count > 0)
-                || statements.Any(stmt => AstSearcher.Contains(stmt, ast => ast is TrapStatementAst, searchNestedScriptBlocks: false))))
+            if (!_compilingTrap
+                && ((traps != null && traps.Count > 0)
+                    || AstSearcher.AnyContains(statements, ast => ast is TrapStatementAst, searchNestedScriptBlocks: false)))
```

Comment:
Yeah there are definitely some other options around there. I was mostly going off how it read here and named the method accordingly. Ultimately this could just be an additional `Contains()` method overload that takes multiple statements if we wanted.

Reply:
I don't know how it is more native for English - ContainsAnyOf() ?

Reply:
ContainsAny() has the same implication as just Contains() whereas AnyContains() implies that each child element is a collection/tree which could contain the desired thing.

I would like a better name, but without getting overly long I'm not sure there are many good options.